process/transactionProcessor.go

```diff
@@ -150,24 +150,72 @@ func (tp *TransactionProcessor) SimulateTransaction(tx *data.Transaction) (*data
 		return nil, err
 	}
 
-	shardID, err := tp.proc.ComputeShardId(senderBuff)
+	senderShardID, err := tp.proc.ComputeShardId(senderBuff)
 	if err != nil {
 		return nil, err
 	}
 
-	observers, err := tp.proc.GetObservers(shardID)
+	observers, err := tp.proc.GetObservers(senderShardID)
+	if err != nil {
+		return nil, err
+	}
+
+	response, err := tp.simulateTransaction(observers, tx)
+	if err != nil {
+		return nil, fmt.Errorf("%w while trying to simulate on sender shard (shard %d)", err, senderShardID)
+	}
+
+	receiverBuff, err := tp.pubKeyConverter.Decode(tx.Receiver)
+	if err != nil {
+		return nil, err
+	}
+
+	receiverShardID, err := tp.proc.ComputeShardId(receiverBuff)
+	if err != nil {
+		return nil, err
+	}
+
+	if senderShardID == receiverShardID {
+		return &data.GenericAPIResponse{
+			Data:  response.Data,
+			Error: response.Error,
+			Code:  response.Code,
+		}, nil
+	}
+
+	// TODO: do we need simulating in receiver shard also for all kind of txs? or just smart contract calls?
+	observersForReceiverShard, err := tp.proc.GetObservers(receiverShardID)
 	if err != nil {
 		return nil, err
 	}
 
+	responseFromReceiverShard, err := tp.simulateTransaction(observersForReceiverShard, tx)
+	if err != nil {
+		return nil, fmt.Errorf("%w while trying to simulate on receiver shard (shard %d)", err, receiverShardID)
+	}
+
+	simulationResult := data.ResponseTransactionSimulationCrossShard{}
+	simulationResult.Data.Result = map[string]data.TransactionSimulationResults{
+		"senderShard":   response.Data.Result,
```

Comment:
Optional: you could have actual properties instead of a map with key-values.

Reply:
This could be helpful, but it still doesn't solve the response consistency for intra-shards vs cross-shard transactions. 
Now we have:
single shard: data -> SimulationResponse
cross shard: data -> map[string]SimulationResponse

Maybe a map of a slice of a struct containing `shard` and `SimulationResponse` ? As in:
single shard: data -> struct{shard: "shard 0", simulation}
cross shard: data -> [] {struct{shard: "shard 0", simulation}, struct{"shard 1", simulation"} 

Where the shard could be a basic uint32 or a string (with possible additional details, as in "shard 0 - receiver shard")